packages/devtools/lib/src/timeline/frame_flame_chart.dart

```diff
@@ -407,3 +430,142 @@ class FlameChartItem {
         colorToCss(selected ? selectedTextColor : defaultTextColor);
   }
 }
+
+class TimelineGrid extends CoreElement {
+  TimelineGrid(this._frameDurationMs, this.flameChartWidth)
+      : super('div', classes: 'flame-chart-grid') {
+    flex();
+    layoutHorizontal();
+    _drawGrid(baseGridInterval);
+  }
+
+  static const baseGridInterval = 150;
+
+  final num _frameDurationMs;
+
+  num zoomLevel = 1;
+
+  num flameChartWidth;
+
+  num get flameChartWidthWithInsets => flameChartWidth + 2 * flameChartInset;
+
+  num interval = baseGridInterval;
+
+  final List<TimelineGridItem> _gridItems = [];
+
+  void _drawGrid(num interval) {
+    // Draw the first grid item since it will have a different width than the
+    // rest.
+    final gridItem = TimelineGridItem(0, flameChartInset, 0);
+    _gridItems.add(gridItem);
+    add(gridItem);
+
+    num left = flameChartInset;
+
+    while (left + interval < flameChartWidthWithInsets) {
+      final timestamp = getTimestampForPosition(left + interval);
```

Comment:
i think this logic would be simpler if we explicitly calculated the timestamp and then determined what its offset is instead of inverting.
That would also avoid some rounding errors. Its ok to add a TODO for that and a follow up CL.

Reply:
I'm not sure I'm following. Are you saying we should do `getTimestampForPosition(left)` and then add an offset based on how wide the interval is? Or calculate how much time each interval should represent, and calculate each timestamp as n * timePerInterval?

Reply:
I'm saying that instead of getting the timestamp for a position it would be simpler to:
1. determine the timestamp we want
2. determine the position for it.
This could mean just tracking a timestamp variable
incrementing it by the time interval each griditem represents with each frame.

Reply:
Ah i see. added TODO.